server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/SetDefaultAction.java

```diff
@@ -60,45 +53,29 @@ public void define(WebService.NewController controller) {
       .setPost(true)
       .setHandler(this);
 
-    setDefault.createParam(PARAM_LANGUAGE)
-      .setDescription("The key of a language supported by the platform. If specified, profileName must be set to select the default profile for the selected language.")
-      .setExampleValue("js")
-      .setPossibleValues(LanguageParamUtils.getLanguageKeys(languages));
-
-    setDefault.createParam(PARAM_PROFILE_NAME)
-      .setDescription("The name of a quality profile. If specified, language must be set. The matching profile will be used as default for the selected language.")
-      .setExampleValue("Sonar way");
+    qProfileWsSupport.createOrganizationParam(setDefault)
+      .setSince("6.4");
 
-    setDefault.createParam(PARAM_PROFILE_KEY)
-      .setDescription("The key of a quality profile. If specified, language and profileName must not be set. The matching profile will be used as default for its language.")
-      .setExampleValue("sonar-way-js-12345");
+    QProfileReference.defineParams(setDefault, languages);
   }
 
   @Override
-  public void handle(Request request, Response response) throws Exception {
-    qProfileWsSupport.checkQProfileAdminPermission();
-
-    String language = request.param(PARAM_LANGUAGE);
-    String profileName = request.param(PARAM_PROFILE_NAME);
-    String profileKey = request.param(PARAM_PROFILE_KEY);
-
-    checkArgument(
-      (!isEmpty(language) && !isEmpty(profileName)) ^ !isEmpty(profileKey), "Either profileKey or profileName + language must be set");
-
-    if (profileKey == null) {
-      profileKey = getProfileKeyFromLanguageAndName(language, profileName);
+  public void handle(Request request, Response response) {
```

Comment:
BTW: This is behaviour is shared with some other web services (though class "QProfileReference")